packages/mom5/package.py

```diff
@@ -44,16 +38,40 @@ class Mom5(MakefilePackage):
         depends_on("oasis3-mct~deterministic", when="~deterministic")
         depends_on("libaccessom2+deterministic", when="+deterministic")
         depends_on("libaccessom2~deterministic", when="~deterministic")
-    with when("@access-esm1.5:access-esm1.6"):
+        depends_on("access-fms")
+        depends_on("access-generic-tracers")
+        depends_on("access-mocsy")
+
+    with when("@access-esm1.5"):
+        depends_on("netcdf-c@4.7.1:")
+        depends_on("netcdf-fortran@4.5.1:")
+        # Depend on "openmpi".
+        depends_on("openmpi")
+        depends_on("oasis3-mct")
+
+    with when("@access-esm1.6"):
         depends_on("netcdf-c@4.7.1:")
         depends_on("netcdf-fortran@4.5.1:")
         # Depend on "openmpi".
         depends_on("openmpi")
         depends_on("oasis3-mct")
+        depends_on("access-fms")
+        depends_on("access-generic-tracers")
+        depends_on("access-mocsy")
 
     phases = ["edit", "build", "install"]
 
-    _platform = "spack"
+    __builds = {
+        "access-om2": {"type": "ACCESS-OM", "gtracers": True},
+        "access-om2-bgc-legacy": {"type": "ACCESS-OM-BGC", "gtracers": False},
+        "access-esm1.5": {"type": "ACCESS-CM", "gtracers": False},
+        "access-esm1.6": {"type": "ACCESS-ESM", "gtracers": True}
+    }
+    # These two variables should be set correctly in edit():
+    __type = "INVALID"
+    __gtracers = "INVALID"
```

Comment:
These to be set to ACCESS-OM2 values when there is no version in the spec. I presume this would work?

```suggestion
    __type = "ACCESS-OM"
    __gtracers = True
```

Alternatively, could change the logic in the `edit` section to default to the ACCESS-OM2 values

Reply:
It's better to set them to invalid values that will fail if there's a problem with processing the data structure.